Mathlib/Order/Interval/Set/Infinite.lean

```diff
@@ -16,15 +16,13 @@ preorder is an infinite type.
 
 variable {α : Type*} [Preorder α]
 
-/-- A nonempty preorder with no maximal element is infinite. This is not an instance to avoid
-a cycle with `Infinite α → Nontrivial α → Nonempty α`. -/
-theorem NoMaxOrder.infinite [Nonempty α] [NoMaxOrder α] : Infinite α :=
+/-- A nonempty preorder with no maximal element is infinite. -/
+instance NoMaxOrder.infinite [Nonempty α] [NoMaxOrder α] : Infinite α :=
   let ⟨f, hf⟩ := Nat.exists_strictMono α
   Infinite.of_injective f hf.injective
 
-/-- A nonempty preorder with no minimal element is infinite. This is not an instance to avoid
-a cycle with `Infinite α → Nontrivial α → Nonempty α`. -/
-theorem NoMinOrder.infinite [Nonempty α] [NoMinOrder α] : Infinite α :=
+/-- A nonempty preorder with no minimal element is infinite. -/
+instance NoMinOrder.infinite [Nonempty α] [NoMinOrder α] : Infinite α :=
```

Comment:
... and also:
```suggestion
/-- A nonempty preorder with no minimal element is infinite.
This instance has lower priority to avoid the cycle `Infinite α → Nontrivial α → Nonempty α` if possible.
-/
instance (priority := 100) NoMinOrder.infinite [Nonempty α] [NoMinOrder α] : Infinite α :=
```